src/azul/service/manifest_service.py

```diff
@@ -2059,6 +2059,23 @@ def create_file(self) -> tuple[str, str | None]:
         replicas = list(self._list_replicas())
         plugin = self.metadata_plugin
         replica_schemas = plugin.verbatim_pfb_schema(replicas)
+        # FIXME: Refactor injection of snapshot ID field into metadata plugin
+        #        https://github.com/DataBiosphere/azul/issues/7411
+        if config.is_anvil_enabled(self.catalog):
+            for replica in replicas:
+                if replica['replica_type'] == 'anvil_dataset':
+                    source_id = replica['source']['id']
+                    replica['contents']['source_datarepo_snapshot_id'] = source_id
+            assert hasattr(plugin, '_pfb_schema_from_anvil_column')
```

Comment:
I don't see the point of this. For one, it is brittle. If the name of the method changes, the assert would start firing because most refactoring tool don't connect the dots. And whether the an assertion fires in line 2069 or an attribute error is raise four lines down in line 2072 doesn't really make a difference as far as I can tell.